urllib3/request.py

```diff
@@ -149,3 +187,24 @@ def request_encode_body(self, method, url, fields=None, headers=None,
         extra_kw.update(urlopen_kw)
 
         return self.urlopen(method, url, **extra_kw)
+
+    def redirect(self, response, method, retries, **kwargs):
+        """
+        Abstracts the redirect process to be used from any :class:`RequestMethods` object
+        """
+        url = kwargs.pop('url', '')
+        redirect_location = urljoin(url, response.get_redirect_location())
+        method = retries.redirect_method(method, response.status)
+        try:
+            pool = kwargs.pop('pool', self)
```

Comment:
I believe `MaxRetryError` is actually what gets raised when we have too many redirects, so this block can raise that exception. Although, I'm not sure, in that case, if the connection still needs to be released or if it would have been closed already by a lower-level routine.